noxfile.py

```diff
@@ -1,9 +1,33 @@
 import nox
 
+
+@nox.session(name="pip-compile", python=["3.11"])
+def pip_compile(session: nox.Session):
+    # .pip-tools.toml was introduced in v7
+    session.install("pip-tools >= 7")
+
+    # Use --upgrade by default unless a user passes -P.
+    args = list(session.posargs)
+    if not any(
+        arg.startswith(("-P", "--upgrade-package", "--no-upgrade")) for arg in args
+    ):
+        args.append("--upgrade")
+
+    session.run(
+        "pip-compile",
+        "--output-file",
+        f"requirements.txt",
+        *args,
+        f"requirements.in",
+    )
+
+
 @nox.session
 def build(session: nox.Session):
     session.install(
-      "-r", "requirements.in",
-      "-c", "requirements.txt",
+        "-r",
+        "requirements.in",
+        "-c",
+        "requirements.txt",
```

Comment:
Looks like a code formatter broke this. It's rather annoying that options and their values aren't paired up. One workaround could be using a syntax like `"--constraint=requirements.txt"` that wouldn't let formatters separate related pieces of information.

Reply:
bah. I failed to spot that. I can add `# fmt: off` `# fmt: on` comments to prevent the formatter from breaking this again.